Sources/MapboxNavigation/WayNameView.swift

```diff
@@ -2,12 +2,161 @@ import Foundation
+     - returns: `true` if operation was successful, `false` otherwise.
+     */
+    @discardableResult
+    private func setupWithLegacy(roadName: String, shieldIcon: UIImage) -> Bool {
+        var currentShieldName: NSAttributedString?, currentRoadName: String?
+        var didSetup = false
+
+        let attachment = ShieldAttachment()
+        let fontSize = frame.size.height / 2.5
+        attachment.image = shieldIcon.withFontSize(font: UIFont.boldSystemFont(ofSize: fontSize),
+                                                   size: frame.size)
+        currentShieldName = NSAttributedString(attachment: attachment)
+
+        if !roadName.isEmpty {
+            currentRoadName = roadName
+            text = roadName
+            didSetup = true
+        }
+
+        if let compositeShieldImage = currentShieldName, let roadName = currentRoadName {
+            let compositeShield = NSMutableAttributedString(string: " \(roadName)")
+            compositeShield.insert(compositeShieldImage, at: 0)
+            attributedText = compositeShield
+            didSetup = true
+        }
+        return didSetup
+    }
+    
+    /**
+     Fills contents of the `WayNameLabel` with the road name and shield icon from the spriteRepository.
+     
+     - parameter roadName: The road name `String` that should be presented on the view.
+     - parameter shieldIcon: The  `UIImage` that represents the current road shield.
+     - returns: `true` if operation was successful, `false` otherwise.
+     */
+    @discardableResult
+    private func setupWithShield(roadName: String, shieldIcon: UIImage) -> Bool {
+        guard let shield = representation?.shield else { return false }
+
+        var currentShieldName: NSAttributedString?, currentRoadName: String?
+        var didSetup = false
+        currentShieldName = roadShieldAttributedText(for: shield.text, textColor: shield.textColor, image: shieldIcon)
+
+        if !roadName.isEmpty {
+            currentRoadName = roadName
+            text = roadName
+            didSetup = true
+        }
+        
+        if let compositeShieldImage = currentShieldName, let roadName = currentRoadName {
+            let compositeShield = NSMutableAttributedString(string: " \(roadName)")
+            compositeShield.insert(compositeShieldImage, at: 0)
+            attributedText = compositeShield
+            didSetup = true
+        }
+        
+        return didSetup
+    }
+    
+    private func shieldColor(from shieldTextColor: String) -> UIColor {
+        switch shieldTextColor {
+        case "black":
+            return .black
+        case "blue":
+            return .blue
+        case "green":
+            return .green
+        case "red":
+            return .red
+        case "white":
+            return .white
+        case "yellow":
+            return .yellow
+        case "orange":
+            return .orange
+        default:
+            return .black
```

Comment:
The text color should match the color scheme of the spritesheet and stylesheet. For example, green is approximately `#00ff00` (depending on color correction and color profile), which is much too green for any visually pleasing map style. For now, we may need to hardcode values in the UI style based on what’s being used in the map style.